qe/drivers.py

```diff
@@ -1913,50 +1880,48 @@ def E(self, psi_coef: Psi_coef) -> Energy:
         return E.item()
 
     @cached_property
-    def psi_external(self):
+    def psi_external_chunk(self):
         # Generate all external (connected) determinants for current CIPSI iteration
-        # Todo this need to be a generator, we cannot store all the connected
-        return Excitation(self.H_i_generator.N_orb).gen_all_connected_determinant(
-            self.H_i_generator.psi_internal
+        # TODO: this need to be a generator, we cannot store all the connected
+        # Return `chunk' of external space given to MPI rank.
+        # Function `gen_all_connected_determinant' gets chunk automatically
+
+        # Compute external determinants + size of external space
+        external_chunk, external_size = Excitation(
+            self.H_i_generator.N_orb
+        ).gen_all_connected_determinant(self.comm, self.H_i_generator.psi_internal)
+        # Save size of external space (used in MPI communications below)
+        self.external_size = external_size
```

Comment:
Where it's used? We should never need so "global" external size, only the local.